bpf/aya-bpf/src/maps/array.rs

```diff
@@ -10,45 +10,47 @@ use crate::{
 
 #[repr(transparent)]
 pub struct Array<T> {
-    def: bpf_map_def,
+    def: UnsafeCell<bpf_map_def>,
     _t: PhantomData<T>,
 }
 
+unsafe impl<T: Sync> Sync for Array<T> {}
```

Comment:
Why do we need to implement `Sync` we don't have multiple threads here

Reply:
`static` vars are assumed to be accessible from multiple threads. eBPF programs in general can be called from multiple threads as well.

Reply:
To be clear: it doesn't matter if the Map itself is thread safe or not because it's actually immutable anyway. But in terms of API design, this static object has to be thread safe, especially T, or you will have races when you have something like `Map<RefCell<T>>` and multiple threads tries to access that.